### PR TITLE
ci: verify gpg signatures of dependencies in CI

### PR DESCRIPTION
**Pull request by pierwill**
Verify signatures of bazel and influxdb2.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Looks like a nice change to me -- thank you @pierwill

**Comment by alamb** [comment]
Any comments @jacobmarble ?

File: docker/Dockerfile.ci [hunk 1]
```diff
@@ -32,17 +32,27 @@ RUN apt-get update \
...
 
-# Install bazel using the binary installer to enable building of flatc in the flatbuffers check
-RUN curl -Lo bazel-4.0.0-installer-linux-x86_64.sh https://github.com/bazelbuild/bazel/releases/download/4.0.0/bazel-4.0.0-installer-linux-x86_64.sh \
-  && chmod +x bazel-4.0.0-installer-linux-x86_64.sh \
-  && ./bazel-4.0.0-installer-linux-x86_64.sh \
-  && rm bazel-4.0.0-installer-linux-x86_64.sh
+# Install bazel using the installer script to enable building of flatc in the flatbuffers check
+ENV BAZEL_VERSION=4.0.0
+ENV BAZEL_DOWNLOAD_BASE="https://github.com/bazelbuild/bazel/releases/download"
+RUN curl https://bazel.build/bazel-release.pub.gpg | gpg --import - \
```
Comment (crepererum) [inline docker/Dockerfile.ci:38]:
```suggestion
RUN curl --proto '=https' --tlsv1.2 -sSf https://bazel.build/bazel-release.pub.gpg | gpg --import - \
```

To ensure:

- use HTTPS for real
- enforce a sufficient TLS version
- silent mode
- show errors if they occur
- fail on HTTP errors

Taken from https://rustup.rs/ . Similar change should be applied to all `curl` invocations.

**Review by crepererum — CHANGES_REQUESTED** [review]
`curl` (esp. during key import) should probably hardened a bit.

**Comment by pierwill** [comment]
git history's gotten a little weird. 😬

**Review by crepererum — APPROVED** [review]

**Comment by alamb** [comment]
> git history's gotten a little weird. 😬
No worries -- we'll squash it all together

**Review by alamb — APPROVED** [review]
I tested this locally and it seems to work great. Thanks @pierwill

**Pull request merged**